### ISSUE TITLE
[BUG] Remove reliance on android.permission.GET_TASKS

### ISSUE THREAD

**Issue by tulioccalazans** [transistorsoft/flutter_background_fetch#284]
**Your Environment**
* Plugin version: 2.8.1
* Platform: Android
* OS version: API 21+

### Expected Behavior
Does not add **android.permission.GET_TASKS** in the merged manifest.

### Actual Behavior
It adds **android.permission.GET_TASKS** in the merged manifest.

### Steps to Reproduce

1. Add the dependency
2. Build the app
3. Inspect the generated manifest in the APK. It'll be asking for the android.permission.GET_TASKS permission
4. Alternatively, go to app settings on the device (Settings -> Apps & notifications -> your app -> permission -> overflow menu -> all permissions), and look at the permissions. It lists **retrieve running apps** as one of the permissions, which sounds really bad to the end user.

### Context

I noticed that this library adds **android.permission.GET_TASKS** as one of the required permissions in the merged manifest. 

getTasks method is deprecated and the scope of the method has been reduced to return only the owner process's tasks and I know that you are only using this permission to find whether our app's main activity is running or not, but usage of **android.permission.GET_TASKS** is usually a sign of spyware like behaviour.

Due to changes in Google's privacy policy, there are some rejections and warnings because of this permission. As shown in the image, the app was accepted, but with issues to be resolved. Google is giving certain deadlines for compliance with its privacy policies, I believe that the deadline for resolving the pending issue shown in the image is coming to an end. More details at: [https://www.termsfeed.com/blog/google-data-safety-form](https://www.termsfeed.com/blog/google-data-safety-form
![email](https://user-images.githubusercontent.com/3017915/164067545-ece1b67e-c49c-4630-b07c-7bea02e316de.jpg)
)

**Comment by christocracy** [comment]
GET_TASKS is required in order for the plugin to know when the MainActivity is terminated so that it can send events to the register ed headless-task.

It is only a harmless warning that can be ignored.

**Comment by christocracy** [comment]
I'm well aware what GET_TASKS is and how this was deprecated several years ago.

This deprecated feature used to return results for *all* apps running on a device (a security hole).  A few years ago, Android scaled-back the results to return only tasks initiated *only by one's own app*, which is precisely what the plugin uses it for (ie the MainActivity).

**Comment by christocracy** [comment]
The `GET_TASKS` permission is required in order to use the method [getRecentTasks](https://developer.android.com/reference/android/app/ActivityManager.html#getRecentTasks(int,%20int))

<img width="886" alt="Screen Shot 2022-04-19 at 3 03 49 PM" src="https://user-images.githubusercontent.com/14999/164077037-46856cb2-ba01-4093-ae90-2df42a7a54b1.png">

Migrating the plugin to use the more recent [`getAppTasks`](https://developer.android.com/reference/android/app/ActivityManager#getAppTasks()) would require a `minSdkVersion 21` (*Android 5*), cutting off *Android 4*, which the plugin current supports.

**Comment by tulioccalazans** [comment]
> The `GET_TASKS` permission is required in order to use the method [getRecentTasks](https://developer.android.com/reference/android/app/ActivityManager.html#getRecentTasks(int,%20int))
> 
> <img alt="Screen Shot 2022-04-19 at 3 03 49 PM" width="886" src="https://user-images.githubusercontent.com/14999/164077037-46856cb2-ba01-4093-ae90-2df42a7a54b1.png">
> 
> Migrating the plugin to use the more recent [`getAppTasks`](https://developer.android.com/reference/android/app/ActivityManager#getAppTasks()) would require a `minSdkVersion 21` (_Android 5_), cutting off _Android 4_, which the plugin current supports.

Thank you, @christocracy !

I'm in contact with google to pass along this information. I believe that the review was not carried out correctly according to the information you gave me.

I hope they don't take too long to respond.

**Comment by christocracy** [comment]
Your posted screenshot about "Invalid Data Safety Section" is not related to GET_TASKS

**Comment by tulioccalazans** [comment]
I investigated why "Apps on device" is pointed as issue as my app doesn't get the list of apps that are installed on the device. 

Google messages are not very clear most of the time...

I believe "Apps on device" is related to "Installed apps" item from the data security section, shown in the image below.
![ps_app_activity](https://user-images.githubusercontent.com/3017915/164083759-e36ecd21-e257-4233-b0ec-00c06fc5ee07.jpg)

Well, let's see what the answer is. Any news, I'll let you know here.

**Comment by christocracy** [comment]
https://support.google.com/googleplay/android-developer/answer/10787469?hl=en

**Comment by christocracy** [comment]
I got a notification about this a few months ago for all three of my published demo apps.

You just need to answer questions about how you manage data you gather from users.

It's a form to fill out in Play Store Console.  Just follow the directions in the link I just posted above.

**Comment by christocracy** [comment]
It is not at all related to background_fetch.

**Comment by tulioccalazans** [comment]
> I got a notification about this a few months ago for all three of my published demo apps.
> 
> You just need to answer questions about how you manage data you gather from users.
> 
> It's a form to fill out in Play Store Console. Just follow the directions in the link I just posted above.

I also did this a few months ago and everything was fine, but suddenly when I uploaded an app update they started questioning me about it.

As my app doesn't get the list of "Apps on device", I got confused and started researching what it could be.

The android.permission.GET_TASKS permission would have been a problem in the past as it managed to do just that. But with her alteration, she is no longer a problem.

Maybe Google forgot about this, as they forgot the information "retrive running apps" in the app description only because it uses the android.permission.GET_TASKS permission.

Well, if it's not related, I hope they can better direct me the reason.
